language_service/Cargo.toml

```diff
@@ -17,3 +17,4 @@ log = { workspace = true }
 miette = { workspace = true }
 qsc = { path = "../compiler/qsc" }
 enum-iterator = { workspace = true }
+regex = "1.9.1"
```

Comment:
Firstly, don't add dependencies directly in the projects, add them to the workspace and reference them in the projects using the pattern you see for the other dependencies above.

Secondly, this package looks pretty big. I just built from main and got a 1.5MB release wasm file. After pulling this PR and building the wasm file is now 2.5MB. That's an unacceptable increase. Please find a cheaper way.

Reply:
Seems like this is a known issue with Rust's regex create: https://github.com/rust-lang/regex/issues/583. They were able to get the size down a bit, but even then it is still large at ~330K.

Reply:
A newer regex_lite crate was recently released by the same author: https://docs.rs/regex-lite/latest/regex_lite/. After swapping in this create, the file size of the release wasm file when up by 50K, which seems significantly better. @billti if this is acceptable, then we can go forward with this. If not, I can work on a hand-rolled solution. Please let me know.

Reply:
Thanks! That's a much bigger improvement, and I think we can live with that. (The crate seems to have little usage currently, but seeing as it's by the same author from the same repo, I'm guessing it's largely the same code/quality).

He does mention this isn't as performant. I assume we are running one evaluation on just the item you are hovering?

Reply:
That's right. It only calls the `parse_doc` once the hover logic has located the appropriate callable definition.